doc/kinsol/guide/source/Mathematics.rst

```diff
@@ -381,44 +381,50 @@ Anderson Acceleration
 The Picard and fixed point methods can be significantly accelerated using Anderson’s method
 :cite:p:`Anderson65, Walker-Ni09, Fang-Saad09, LWWY11`. Anderson acceleration can be formulated as follows:
 
-1. Set :math:`u_0 =` an initial guess and :math:`m \ge 1`
+1. Set :math:`u_0 =` an initial guess, :math:`m \ge 1`, and :math:`m_0 = 0`
 
 2. Set :math:`u_1 = G(u_0)`
 
 3. For :math:`n = 1, 2,...` until convergence do:
 
-   a.  Set :math:`m_n = \min\{m,n\}`
+   a.  Set :math:`m_n = \min\{m,n,m_{n-1} + 1\}`
 
    b.  Set :math:`F_{n} = (f_{n-m_n}, \ldots, f_n)`, where :math:`f_i=G(u_i)-u_i`
 
-   c.  Determine :math:`\alpha^{(n)} = (\alpha_0^{(n)}, \ldots, \alpha_{m_n}^{(n)})` that solves
+   c.  Optionally, update :math:`m_n` to remove some number of the left-most
```

Comment:
Not sure if this is what's actually done, but it seems wasteful to construct `F_{n}` only then to remove some columns. I.e., should b and c be switched?

Reply:
Some approaches for selecting the depth need to examine the system that will be solved for the acceleration weights. So we have to add a new column to `F_{n}` in this step before selecting the depth. The requires orthogonalizing the new vector against the existing vectors in `F_{n}` but this is usually a (relatively) small cost.